### PR TITLE
enhance: doc parser support schema example

### LINKED ISSUE #668: doc parser: support parse the example section of schema
## Enhancement

the doc parser now only parse the description of schema and schema attribute, and the `Examples` section of the schema is not parsed to the doc.

### related interfaces:

1. the `SchemaType` now contains a String type schema `doc` field, but the examples field is missing.

```rust
pub struct SchemaType {
    /// The schema name.
    pub name: String,
    /// The schema definition package path
    pub pkgpath: String,
    /// The schema definition file path.
    pub filename: String,
    /// The schema definition document string.
    pub doc: String,                                                           // could be a `kclvm_sema::resolver::doc::Doc` type
    /// Indicates whether the schema is a type of a instance or
    /// a type (value). Besides, it is necessary to distinguish
    /// between a type instance and a type value, such as the following code:
    /// ```no_check
    /// # `Person` in `schema Person` is a type and it is not a schema instance.
    /// schema Person:
    ///     name: str
    ///
    /// # `person` is a schema instance.
    /// person = Person {name = "Alice"}
    /// # `person` is a schema instance used in the value expression.
    /// name = person.name
    /// # `Person` in `persons: [Person]` is a type, `Person` in `Person.instances()`
    /// # is a type value, and they are not schema instances.
    /// persons: [Person] = Person.instances()
    /// ```
    pub is_instance: bool,
    /// Indicates whether it is a schema mixin.
    pub is_mixin: bool,
    /// Indicates whether it is a schema protocol.
    pub is_protocol: bool,
    /// Indicates whether it is a rule.
    pub is_rule: bool,
    /// Base schema.
    pub base: Option<Box<SchemaType>>,
    /// Protocol schema.
    pub protocol: Option<Box<SchemaType>>,
    /// Schema Mixins.
    pub mixins: Vec<SchemaType>,
    /// Schema attributes.
    pub attrs: IndexMap<String, SchemaAttr>,
    /// Schema function type.
    pub func: Box<FunctionType>,
    /// Schema index signature.
    pub index_signature: Option<Box<SchemaIndexSignature>>,
    /// Schema decorators including self and attribute decorators.
    pub decorators: Vec<Decorator>,
}
```

2. the `Doc` and `Example`:
```
#[derive(Debug, PartialEq, Eq, Clone)]
pub(crate) struct Doc {
    pub summary: String,
    pub attrs: Vec<Attribute>,
    pub examples: Vec<Example>,            // add a `examples` field to Doc type
}
```
```
pub struct Example {
    /// Short description for the example.
    pub summary: String,
    /// Long description for the example.
    pub description: String,
    /// Embedded literal example.
    pub value: Option<String>,
}
```

### PR DESCRIPTION
**Pull request by amyXia1994**
Close #668

#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [X] Y 

#### 2. What is the scope of this PR (e.g. component or file name):

kclvm/api/src/service/ty.rs
kclvm/sema/src/resolver/doc.rs
kclvm/sema/src/resolver/global.rs
kclvm/sema/src/resolver/mod.rs
kclvm/sema/src/ty/mod.rs

kclvm/spec/gpyrpc/gpyrpc.proto

kclvm/api/src/testdata/get-schema-type-mapping.json
kclvm/api/src/testdata/get-schema-type-mapping.response.json
kclvm/sema/src/resolver/test_data/doc.k

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

This PR support parsing the `Examples` section from the schema docstring, and exposes the examples field from the `KclvmService.GetSchemaTypeMapping` API

Note: The `Example.value` field of the `KclvmService.GetSchemaTypeMapping` API changed to string type.

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [X] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [X] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

### CODE REVIEW

**Comment by amyXia1994** [comment]
@Peefy 

I suggest to change the type of `grpc.Example.value` to `String`, since the example code snippet of the schema can only be represented in plain text. And currently, if we keep using `prost_wkt_types::Any` as the example value, the corresponding json representation of the `Server` schema's example in the `kclvm/api/src/testdata/exec-program-with-include-schema-type-path.response.json` file will be like:
```
{
    "@type":"type.googleapis.com/example.StringValue",
    "value":[109,121,67,117,115,116,111,109,65,112,112,32,61,32,65,112,112,67,111,110,102,105,103,117,114,97,116,105,111,110,32,123,10,32,32,32,32,110,97,109,101,32,61,32,34,99,111,109,112,111,110,101,110,116,78,97,109,101,34,10,125]
}
```, which is unreadable manually.

File: kclvm/sema/src/resolver/doc.rs [hunk 1]
```diff
@@ -274,19 +274,29 @@ pub(crate) fn parse_doc_string(ori: &String) -> Doc {
...
     pub summary: String,
     pub attrs: Vec<Attribute>,
+    pub examples: HashMap<String, Example>,
 }
 
 impl Doc {
-    fn new(summary: String, attrs: Vec<Attribute>) -> Self {
-        Self { summary, attrs }
+    fn new(summary: String, attrs: Vec<Attribute>, examples: HashMap<String, Example>) -> Self {
+        Self { summary, attrs, examples: examples }
```
Comment (Peefy) [inline kclvm/sema/src/resolver/doc.rs:299]: Just write the code. `Self { summary, attrs, examples }`

File: kclvm/api/src/testdata/get-schema-type-mapping.json [hunk 2]
```diff
@@ -1,4 +1,4 @@
...
...st practice of Kubernetes.\n\n    Attributes\n    ----------\n    workloadType: str, default is \"Deployment\", required\n        Use this attribute to specify which kind of long-running service you want.\n        Valid values: Deployment, CafeDeployment.\n        See also: kusion_models/core/v1/workload_metadata.k.\n    name: str, required\n        A Server-level attribute.\n        The name of the long-running service.\n        See also: kusion_models/core/v1/metadata.k.\n    labels: {str:str}, optional\n        A Server-level attribute.\n        The labels of the long-running service.\n        See also: kusion_models/core/v1/metadata.k.\n\n    Examples\n    ----------------------\n    myCustomApp = AppConfiguration {\n        name = \"componentName\"\n    }\n    \"\"\"\n    workloadType: str = \"Deployment\"\n    @info(\"name\", key=\"value\")\n    name: str\n    labels?: {str:str}\n    useCustomizeLables: True | False\n    containers: [Container]\n\nschema Container:\n    \"\"\"Container is the common user interface for long-running services.\n\n    Attributes\n    ----------\n    name: str, required\n        The name of the long-running container.\n    \"\"\"\n    name: str\n"
```
Comment (Peefy) [inline kclvm/api/src/testdata/get-schema-type-mapping.json:3]: Pls add more description about this PR and run `make fmt` to format the whole source code.

Besides, U can change the API protobuf definition directly about the `example` field. When the compilation and testing are completed, the structure on the rust side will automatically change.

File: kclvm/sema/src/resolver/doc.rs (hunk 1 above)
Comment (amyXia1994) [reply kclvm/sema/src/resolver/doc.rs]: fixed, ptal

File: kclvm/api/src/testdata/get-schema-type-mapping.json (hunk 2 above)
Comment (amyXia1994) [reply kclvm/api/src/testdata/get-schema-type-mapping.json:3]: - code re-formatted
- PR description updated
ptal~

**Review by Peefy — APPROVED** [review]
LGTM

**Pull request merged**